### PR TITLE
refactor(router): Update integration tests to cover navigation and history API

### PR DESCRIPTION
**Pull request by atscott**
This commit updates the router integration tests to cover both the classic History and the new Navigation API. There is more work to be done here, but this commit works to prove the efficacy of the `FakeNavigation` implementation.

### CODE REVIEW

File: packages/common/testing/src/mock_platform_location.ts [hunk 1]
```diff
@@ -250,3 +252,82 @@ export class MockPlatformLocation implements PlatformLocation {
     }
   }
 }
+
+/**
+ * Mock implementation of URL state.
+ */
+@Injectable()
+export class NavigationBasedMockPlatformLocation implements PlatformLocation {
```
Comment (tbondwilkinson) [inline packages/common/testing/src/mock_platform_location.ts:260]: Nit: Naming?

I tend to think that this is more of a "Fake" than a "Mock"

So maybe, "FakeNavigationPlatformLocation"?

**Review by tbondwilkinson — APPROVED** [review]

**Comment by atscott** [comment]
This PR was merged into the repository by commit eb2e879a008adebe11d63f42799570ba4f053c52.

**Comment by atscott** [comment]
This seems blocked by the API extractor because of the private export referencing `Navigation` types.

// cc @devversion

**Comment by devversion** [comment]
@atscott is `Navigation` (the `implements `Navigation` part) referring to a browser global of `dom.d.ts`?

**Comment by atscott** [comment]
> @atscott is `Navigation` (the `implements Navigation` part) referring to a browser global of `dom.d.ts`? seems like this fails to resolve.

It is a browser global provided by `@types/dom-navigation`. I think this came up before with `@types/dom-view-transitions` and we decided to just duplicate the one interface we needed to punt the problem down the road a bit. That's a bit more difficult here since there are a _lot_ more types we'd need to duplicate. It's still possible but it would be good to fix this if we can.

**Comment by devversion** [comment]
I can look more, but maybe the type package is not mentioned in the API golden rule via the "types" attributr

**Comment by atscott** [comment]
@devversion That would be helpful. I wasn't able to figure it out.

**Comment by devversion** [comment]
@atscott I tried reproducing this locally, but not sure if you are seeing the same, but I don't even see these failures on my end?

**Comment by atscott** [comment]
@devversion it’s the npm_package target that’s failing to build on CI it seems. We fixed it locally with the ///references directive but it doesn’t seem to have fixed it on CI

**Comment by atscott** [comment]
caretaker note: Tom's approval should be sufficient here. build file updates are trivial and do not need dev-infra review

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 15c48113c23d95a622bcb798f62e33e751509cb9.

**Pull request closed without merging**